content/tutorial/tutorial.md

```diff
@@ -513,63 +519,64 @@ class Board extends React.Component {
 
-#### Detecting Changes {#detecting-changes}
+#### परिवर्तन का पता लगाना {#detecting-changes}
 
-Detecting changes in mutable objects is difficult because they are modified directly. This detection requires the mutable object to be compared to previous copies of itself and the entire object tree to be traversed.
+मयुटेबल वस्तुओं में परिवर्तन का पता लगाना मुश्किल है क्योंकि वे सीधे संशोधित होते हैं। इसको पता लगाने के लिए परिवर्तनशील वस्तु की आवश्यकता होती है जिसकी पिछली प्रतियों की तुलना की जाए और पूरे ऑब्जेक्ट ट्री को ट्रैवर्स किया जाए।
 
-Detecting changes in immutable objects is considerably easier. If the immutable object that is being referenced is different than the previous one, then the object has changed.
+अपरिवर्तनीय वस्तुओं में परिवर्तनों का पता लगाना काफी आसान है। यदि संदर्भित की जा रही अपरिवर्तनीय ऑब्जेक्ट पहले से भिन्न है, तो वह वस्तु बदल गई है।
 
-#### Determining When to Re-Render in React {#determining-when-to-re-render-in-react}
+#### React में री-रेंडर कब करना है यह निर्धारित करना {#determining-when-to-re-render-in-react}
 
-The main benefit of immutability is that it helps you build _pure components_ in React. Immutable data can easily determine if changes have been made which helps to determine when a component requires re-rendering.
+अपरिवर्तनीयता का मुख्य लाभ यह है कि यह आपको React में _pure components_ बनाने में मदद करता है। अपरिवर्तनीय डेटा आसानी से निर्धारित कर सकते हैं कि क्या परिवर्तन किए गए हैं जो यह निर्धारित करने में मदद करता है कि एक कौम्पोनॅन्ट को फिर से री-रेंडरिंग की आवश्यकता कब होगी।
 
-You can learn more about `shouldComponentUpdate()` and how you can build *pure components* by reading [Optimizing Performance](/docs/optimizing-performance.html#examples).
+[ऑप्टिमिज़िंग परफॉरमेंस](/docs/optimizing-performance.html#examples) पढ़ के आप `shouldComponentUpdate()` के बारे में और जानकारी प्राप्त कर सकते हैं और *pure कौम्पोनॅन्ट* बनाने के बारे में जान सकते हैं।
 
-### Function Components {#function-components}
+### फंक्शन कंपोनेंट्स {#function-components}
 
-We'll now change the Square to be a **function component**.
+अब हम स्क्वायर को **फंक्शन कंपोनेंट्स** में बदल देंगे।
 
-In React, **function components** are a simpler way to write components that only contain a `render` method and don't have their own state. Instead of defining a class which extends `React.Component`, we can write a function that takes `props` as input and returns what should be rendered. Function components are less tedious to write than classes, and many components can be expressed this way.
+React में, **फंक्शन कंपोनेंट्स** उन कौम्पोनॅन्ट को लिखने का एक सरल तरीका है जिनमें केवल एक `render` मेथड शामिल है और उनकी अपनी स्थिति नहीं है। एक स्क्वायर को परिभाषित करने के बजाय जो `React.Component` का विस्तार करता है, हम एक फ़ंक्शन लिख सकते हैं जो `props` को इनपुट के रूप में लेता है और जो रेंडर किया जाना चाहिए वह देता है। 
```

Comment:
स्थिति -> state?

Reply:
एक स्क्वायर को परिभाषित करने के बजाय जो `React.Component` का विस्तार करता है -> revisit?

Reply:
> एक स्क्वायर को परिभाषित करने के बजाय जो `React.Component` का विस्तार करता है -> revisit?

```
एक class को परिभाषित करने के बजाय जो React.Component को extend करता है
```
?

Reply:
the complete sentence still feels out of context

Reply:
```
React में, **फंक्शन कौम्पोनॅन्टस** उन कौम्पोनॅन्ट को लिखने का एक सरल तरीका है जिनमें केवल एक `render` मेथड शामिल है और उनका अपना state नहीं है। हम एक फंक्शन लिख सकते है जो एक class को परिभाषित करने के बजाय `React.Component` को extend करेगा और यह कौम्पोनॅन्ट props इनपुट लेगा और जो render करना है वो रिटर्न करेगा। 
फ़ंक्शन कौम्पोनॅन्टस classes की तुलना में लिखना आसान है, और कई कौम्पोनॅन्टस को इस तरह से व्यक्त किया जा सकता है।
```

Reply:
I have made all the changes, just this is pending, I have not pushed anything, once I commit the latest changes all these fixes will appear with it

Reply:
> ```
> React में, **फंक्शन कौम्पोनॅन्टस** उन कौम्पोनॅन्ट को लिखने का एक सरल तरीका है जिनमें केवल एक `render` मेथड शामिल है और उनका अपना state नहीं है। हम एक फंक्शन लिख सकते है जो एक class को परिभाषित करने के बजाय `React.Component` को extend करेगा और यह कौम्पोनॅन्ट props इनपुट लेगा और जो render करना है वो रिटर्न करेगा। 
> फ़ंक्शन कौम्पोनॅन्टस classes की तुलना में लिखना आसान है, और कई कौम्पोनॅन्टस को इस तरह से व्यक्त किया जा सकता है।
> ```

thiks looks good